templates/purge-registry.py.j2

```diff
@@ -0,0 +1,142 @@
+#!/usr/bin/env python
+
+from argparse import ArgumentParser, RawTextHelpFormatter
+from os import environ
+import datetime
+import json
+import logging
+import requests
+
+HELP_DEFINITION='''
+Script to initiate Gitea Image registry organization
+'''
+HELP_EXAMPLE='''
+Example:
+\n\t./purge-registry.py \
+  --url https://example.com/gitea \
+  --user admin \
+  --password password123 \
+  --retention 5
+'''
+
+headers= {
+    'Content-type': 'application/json'
+}
+
+DATE_FORMAT = '%Y-%m-%dT%H:%M:%SZ'
+
+class Artifact:
+
+    def __init__(self, name, owner, creation_date, version):
+        self.name           = name
+        self.owner          = owner
+        self.creation_date  = datetime.datetime.strptime(creation_date, DATE_FORMAT)
+        self.version        = version
+ 
+    def should_be_purge(self, retention_time):
+        if (self.creation_date + datetime.timedelta(days=retention_time)) < datetime.datetime.now() :
+            return True
+        else:
+            logging.info('The artifact %s is %s old' % (self.name, abs((datetime.datetime.now() - self.creation_date))))
+            return False
+
+def get_call(url, basic):
+    logging.debug('HTTP GET call to %s' % url)
+    res=requests.get(url, auth=basic)
+    res.raise_for_status()
+    return res
```

Comment:
You are repeating yourself too much. This needs to be a `GiteaAPI` class that has thin-wrappers around `_request()`.

Like this:
https://github.com/status-im/infra-utils/blob/34e5d068a46c408426b7b6c4182f1ab828632528/ethereum/validator_check.py#L12-L34